templates/api.hbs

```diff
@@ -1,17 +1,39 @@
 {{>header}}
 
-import { Express } from 'express'
+import { Express, Request, Response } from 'express'
 import passport from 'passport'
+import multer from 'multer'
 import * as t from './types'
 import * as v from '../../validation'
+{{#if (hasOperationSupportingMultipart operations)}}
+import * as f from '../../impl/helpers/{{lowerCase name}}MultipartHelper'
+{{/if}}
 import { Api } from '../../models'
 
 export default function(app: Express, impl: t.{{className name}}Api) {
 	{{#each operations}}
 	app.{{lowerCase httpMethod}}(
 		{{{stringLiteral (pathTemplate fullPath)}}},
 		{{>frag/apiSecurityRequirements}}
-		function (req, res) {
+		{{#if (hasOperationSupportingMultipart .)}}
+		{{#with (getFileProperties requestBody.defaultContent.schema.properties)}}
+		(req, res, next) => {
+			f.upload.fields([{{#each this}}{ name: {{{stringLiteral name}}}, maxCount: {{#if maxCount}}{{{maxCount}}}{{else}}1{{/if}} }{{#hasMore}}, {{/hasMore}}{{/each}}])(req, res, (error) => {
+				if (error?.code === 'LIMIT_UNEXPECTED_FILE') {
+					res.status(400).send(`You have exceeded the maximum number of files allowed for the field '${error.field}'`)
```

Comment:
Let's include the maximum number of files in the error message.

```typescript
res.status(400).send(`The field '${fieldName}' allows a maximum of {{#if maxCount}}{{{maxCount}}}{{else}}1{{/if}} file(s)`)
```

This also syncs up the message with the minimum. I preferred the minimum message style as "you" is a bit personal for an API message.

Reply:
Bit of an issue with this approach: the error returned by Multer does not contain the max number of files allowed for a field.

Reply:
I've devised a solution for this.